### PR TITLE
Adds eslint to build-chain.

### PR DESCRIPTION
**Pull request by tylercecil**
I have NO preferences for linting rules, and am not super aware of a populare typescript style guide. If you have opinions, suggest away!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,9 @@
   "scripts": {
     "test": "jest",
     "build": "webpack-cli",
-    "watch": "webpack-cli -w"
+    "watch": "webpack-cli -w",
+    "lint": "eslint . --ext .ts",
+    "lint-and-fix": "eslint . --ext .ts --fix"
```
Comment (duncgibbs) [inline package.json:10]: Semicolons are usually the way I see alternate commands built. i.e. `"lint:fix"`

**Comment by tylercecil** [comment]
@duncgibbs do these changes look good?

File: .circleci/config.yml [hunk 2]
```diff
@@ -10,7 +10,9 @@ jobs:
       - node/with-cache:
           steps:
             - run: npm install
+            - run: npm run-script build
             - run: npm test
+            - run: npm run-script lint
```
Comment (duncgibbs) [inline .circleci/config.yml:15]: I would do `npm run build` and `npm run lint`

Comment (tylercecil) [reply .circleci/config.yml]: Done.

**Comment by tylercecil** [comment]
Added `singleQuote`. Merging.

**Pull request merged**